aten/src/ATen/cuda/CUDAGraphsUtils.cuh

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include <ATen/CUDAGeneratorImpl.h>
+#include <ATen/cuda/CUDAEvent.h>
+#include <c10/core/StreamGuard.h>
+#include <c10/cuda/CUDAGuard.h>
+
+namespace at {
+namespace cuda {
+namespace philox {
+
+// We can't write a __device__ function in CUDAGeneratorImpl.h, because it's in ATen.
+// Also, whatever call unpacks PhiloxCudaState in consumer kernels must be inlineable.
+// Easiest thing that comes to mind is, define a free function here, in ATen/cuda.
+// Any cuda consumer can include this header.
+__device__ __forceinline__ std::tuple<uint64_t, uint64_t>
+unpack(at::PhiloxCudaState arg) {
+  if (arg.captured_) {
+    return std::make_tuple(arg.seed_, *(arg.offset_.ptr) + arg.offset_intragraph_);
+  } else {
+    return std::make_tuple(arg.seed_, arg.offset_.val);
+  }
+}
+
+} // namespace philox
+
+inline int currentStreamCaptureStatus() {
+  #if defined(CUDA_VERSION) && CUDA_VERSION > 11000
+  // protects against enum cudaStreamCaptureStatus implementation changes
+  // some compilers seem not to like static_assert without the messages.
+  static_assert(int(cudaStreamCaptureStatusNone) == 0,
+                "unexpected int(cudaStreamCaptureStatusNone) value");
+  static_assert(int(cudaStreamCaptureStatusActive) == 1,
+                "unexpected int(cudaStreamCaptureStatusActive) value");
+  static_assert(int(cudaStreamCaptureStatusInvalidated) == 2,
+                "unexpected int(cudaStreamCaptureStatusInvalidated) value");
```

Comment:
Bloop. Why not just define a local enum for these values for portability?

Reply:
Changed it to an enum, but I'm not sure it's exactly what you meant.